businessLogicLayer/Features/PricesNearMe/SubServices/StationDataAPI.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  StationDataAPI.swift
+//  fuelPrices
+//
+//  Created by Robson Harrison on 25/04/2024.
+//
+
+import Foundation
+
+class StationDataAPI: ObservableObject {
+    
+    // MARK: - Error Codes
+    
+    enum StationDataAPIError: Error {
+        case noDataRecieved
+        case networkingError(Error)
+        case jsonDecodingError
+    }
+    
+    // MARK: - Network Calls
+    
+    func requestStationData(from url: URL, completion: @escaping (Result<ResponseData, StationDataAPIError>) -> Void) {
```

Comment:
I think this is cool. It means the the API only has knowledge of how to make one network call. 

So the advantage is that now we can use it for one call, or two, or how-ever-many we want. It's not the responsibility of this class to restrict itself to loops etc. 👍